pkg/kubectl/cmd/apply_view_last_applied.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/ghodss/yaml"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/kubernetes/pkg/kubectl"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+)
+
+type ViewLastAppliedOptions struct {
+	FilenameOptions              resource.FilenameOptions
+	Selector                     string
+	LastAppliedConfigurationList []string
+	OutputFormat                 string
+	Factory                      cmdutil.Factory
+	Out                          io.Writer
+	ErrOut                       io.Writer
+}
+
+var (
+	applyViewLastAppliedLong = templates.LongDesc(`
+		View the latest last-applied-configuration annotations by type/name or file.
+
+		The default output will be printed to stdout in YAML format. One can use -o option
+		to change to output destination.`)
```

Comment:
this message I copied from here: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/convert.go#L46, as a non native speaker, I'm not sure if I copied from the right place ? @pwittrock